### PR TITLE
feat(playground): add example placeholder

### LINKED ISSUE #3652: Make the playground example a placeholder
This would allow the user to enter their code right away without having to delete the example first. Plus, it would make the playground easier to use on mobile — it’s really hard to delete large amounts of content in the code editor on an iOS device.

Also, would it be possible to have different examples for each language? The JS one doesn’t work very well for CSS and Markdown.

(ref https://github.com/prettier/prettier/pull/3573#issuecomment-354602726)

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #3652

Preview: https://deploy-preview-3657--prettier.netlify.com/playground/
  
[Feel free to push examples to this PR directly as I'm not good at making examples.](https://github.com/prettier/prettier/pull/3657#discussion_r159911212) 😅

examples:
- [x] JavaScript (babylon)
- [ ] JavaScript (flow)
- [x] TypeScript
- [ ] CSS
- [ ] Less
- [ ] SCSS
- [ ] JSON
- [ ] GraphQL
- [ ] Markdown

### CODE REVIEW

File: website/static/playground.js [hunk 1]
```diff
@@ -370,3 +352,34 @@ function showTooltip(elem, text) {
...
+        "    { (silent)",
+        '      ? "."',
+        '      : "!"}',
+        "",
+        "    </div>;",
+        "",
+        "}"
+      ].join("\n");
+    default:
+      return "";
```
Comment (ikatyang) [inline website/static/playground.js]: Feel free to push examples to this PR directly as I'm not good at making examples. 😅

File: website/static/playground.js [hunk 2]
```diff
@@ -370,3 +352,56 @@ function showTooltip(elem, text) {
...
+        "      }",
+        "  get bar() {",
+        "    return [  1,",
+        "",
+        "      2, 3,",
+        "    ]",
+        "  }",
+        "}",
+        "",
+        "type RequestType = 'GET' | 'HEAD' | 'POST' | 'PUT' | 'OPTIONS' | 'CONNECT' | 'DELETE' | 'TRACE'"
```
Comment (j-f1) [inline website/static/playground.js]: Is this example too poorly formatted to be believable?

**Comment by j-f1** [comment]
The playground seems to be not working.

**Comment by ikatyang** [comment]
Ref: the playground error https://github.com/prettier/prettier/pull/3624#issuecomment-355595245

**Comment by azz** [comment]
Looks like it crashes on my phone (Chromium).

**Comment by azz** [comment]
That should be easy to fix, just mock out process.binding at the top of worker.js

**Comment by j-f1** [comment]
```
TypeError: path.resolve is not a function
```

**Comment by j-f1** [comment]
```
TypeError: path.resolve is not a function
    at Function.module.exports.sync (https://deploy-preview-3657--prettier.netlify.com/lib/index.js:298:1631)
    at Function.module.exports.sync (https://deploy-preview-3657--prettier.netlify.com/lib/index.js:446:1450)
    at loadPlugins (https://deploy-preview-3657--prettier.netlify.com/lib/index.js:1360:674)
    at normalize$2 (https://deploy-preview-3657--prettier.netlify.com/lib/index.js:1746:128)
    at Object.format (https://deploy-preview-3657--prettier.netlify.com/lib/index.js:2448:286)
    at formatCode (https://deploy-preview-3657--prettier.netlify.com/worker.js:119:21)
    at self.onmessage (https://deploy-preview-3657--prettier.netlify.com/worker.js:69:19)
```

**Comment by vjeux** [comment]
This is pretty cool :)

File: website/static/worker.js [hunk 3]
```diff
@@ -5,20 +5,30 @@
...
 // eslint-disable-next-line
-module$1 = module = path = os = crypto = {};
+module$1 = module = os = crypto = {};
 self.fs = { readFile: function() {} };
 // eslint-disable-next-line no-undef
 os.homedir = function() {
   return "/home/prettier";
 };
-self.process = { argv: [], env: { PRETTIER_DEBUG: true }, version: "v8.5.0" };
+self.process = {
```
Comment (azz) [inline website/static/worker.js]: This stuff is getting a bit out of hand. I wonder if using browserify or rollup-plugin-node-builtins might be better?

Comment (ikatyang) [reply website/static/worker.js]: babelify... 😰

image

**Comment by ikatyang** [comment]
It works! 🎉

**Comment by lydell** [comment]
Also feel free to replace my current JS example with something better. I'm not super happy with it. (If you come up with something really good, we might even want to put it in https://prettier.io/docs/en/index.html – IMO that example isn't super nice either.)

**Comment by azz** [comment]
That example dates _way_ back to the [original blog post](http://jlongster.com/A-Prettier-Formatter), it has historical significance :wink:

File: website/static/worker.js (hunk 3 above)
Comment (duailibe) [reply website/static/worker.js]: We should find a way to remove those dependencies for the browser bundle actually

**Comment by duailibe** [comment]
Nice! I've thought of doing in another way after #3573, removing the text after the 1st click or something and it would continue to be highlighted.

**Comment by ikatyang** [comment]
I'd like to merge this PR and open a <kbd>good first issue</kbd> for `examples`, thoughts?

**Comment by ikatyang** [comment]
Opened #3667.

**Pull request merged**